app/controllers/network_routes_controller.go

```diff
@@ -486,16 +460,43 @@ func (nrc *NetworkRoutingController) getClusterIps() ([]string, error) {
 	return clusterIpList, nil
 }
 
-func (nrc *NetworkRoutingController) getExternalIps() ([]string, error) {
+func (nrc *NetworkRoutingController) getExternalIps(verify_endpoints bool) ([]string, error) {
 	externalIpList := make([]string, 0)
 	for _, svc := range watchers.ServiceWatcher.List() {
+		if verify_endpoints {
+			if svc.Spec.ExternalTrafficPolicy == v1core.ServiceExternalTrafficPolicyTypeLocal {
+				nodeHasEndpoints, err := nrc.nodeHasEndpointsForService(svc)
+				if err != nil {
+					glog.Errorf("error determining if node has endpoints for svc: %q error: %v", svc.Name, err)
+					continue
+				}
+
+				if !nodeHasEndpoints {
+					for _, externalIP := range svc.Spec.ExternalIPs {
+						err := nrc.UnadvertiseClusterIp(externalIP)
```

Comment:
I don't think it makes sense to call `UnadvertiseClusterIp` in `getExternalIps()`. WDTY?